### PR TITLE
list

### PR DESCRIPTION
**Pull request by prabhjots**
Please answer the following questions and leave the below in as part of your PR.

- [X] This PR corresponds to an [issue with a clear problem statement](https://github.com/babashka/babashka/blob/master/doc/dev.md#start-with-an-issue-before-writing-code).

- [X] This PR provides tests for code additions and changes

#88

### CODE REVIEW

**Comment by borkdude** [comment]
@prabhjots Looks good. I think we can now support `(conj nil 1 2 3)` to return a list as well, like in Clojure.

File: core.js [hunk 1]
```diff
@@ -147,34 +160,37 @@ export function conj(...xs) {
   let [o, ...rest] = xs;
 
   if (o === null || o === undefined) {
```
Comment (borkdude) [inline core.js:162]: In Clojure,  `(conj)` returns `[]`, but `(conj nil)` returns `()`.

Comment (prabhjots) [reply core.js:162]: do we want to do the same thing here

Comment (borkdude) [reply core.js:162]: I think so, it's a documented in Clojure, not an unintentional edge case

File: test/clava/compiler_test.cljs [hunk 2]
```diff
@@ -346,7 +346,7 @@
 
 (deftest conj-test
   (testing "corner cases"
-    (is (eq '(), (jsv! '(conj))))
+    (is (eq [], (jsv! '(conj))))
```
Comment (borkdude) [inline test/clava/compiler_test.cljs:349]: eq will compare [] and () successfully, so you might want to check here if the returned object is an array

Comment (prabhjots) [reply test/clava/compiler_test.cljs:349]: fair point. Do you think I can do the following?

```clojure
    (is (= true, (jsv! '(vector? (conj)))))
    (is (= true, (jsv! '(list? (conj nil)))))
```

I will need to add the vector? for that. But that should be easy.

Comment (borkdude) [reply test/clava/compiler_test.cljs:349]: Yep, sounds good
